pandas/core/indexes/base.py

```diff
@@ -4765,7 +4766,7 @@ def _maybe_promote(self, other):
                 return self.astype("object"), other.astype("object")
         return self, other
 
-    def groupby(self, values) -> Dict[Hashable, np.ndarray]:
+    def groupby(self, values) -> PrettyDict:
```

Comment:
now PrettyDict is generic can do
```suggestion
    def groupby(self, values) -> PrettyDict[Hashable, np.ndarray]:
```